### DISCUSSION TITLE
Why are there no `areNotDeclaredInClassesThat()` methods?

### DISCUSSION THREAD

**Discussion by smsajid** [TNG/ArchUnit#1218] [General]
I was searching for a API `areNotDeclaredInClassesThat()` as a negation for the existing API `areDeclaredInClassesThat()` but could not find it.
Is there a specific reason this is not available?

**Comment by codecholeric** [comment]
I think the only reason is that nobody added it so far 😉 But also, the whole API handling is a little inconsistent 😬 Because, on some syntax elements there are these negations, on some are not.
There are definitely ways already to achieve what you need already though... But if you want to add it to the API, I'd also be open for a PR 🙂 
Altogether, if I'd start over today I would design the API differently. Probably no negation starters and just one generic `not()` method that negates the following predicate in a well-defined way 🤷  But, as always with released APIs, hard to change 😬

**Reply by smsajid** [reply]
Thanks for your reply. I have implemented the workaround to get the same behavior. I also prefer the `not()` API.
